plugins/providers/bigquery/provider.go

```diff
@@ -175,7 +176,11 @@ func (p *Provider) GetResources(pc *domain.ProviderConfig) ([]*domain.Resource,
 					table.ProviderURN = pc.URN
 					children[i] = table
 				}
-				dataset.Children = children
+				if containsString(resourceTypes, ResourceTypeDataset) {
+					dataset.Children = children
+				} else {
+					resources = append(resources, children...)
+				}
```

Comment:
Can you add a test case which reflects this fix.

Reply:
sure, I am working on it.